azure/hostedservice/client.go

```diff
@@ -18,24 +18,34 @@ const (
 	azureDeploymentURL                = "services/hostedservices/%s/deployments/%s"
 	deleteAzureDeploymentURL          = "services/hostedservices/%s/deployments/%s?comp=media"
 
-	invalidDnsLengthError  = "The DNS name must be between 3 and 25 characters."
-	paramNotSpecifiedError = "Parameter %s is not specified."
+	errParamNotSpecified = "Parameter %s is not specified."
+	errInvalidDnsLength  = "The DNS name must be between 3 and 25 characters."
 )
 
-func CreateHostedService(dnsName, location string, reverseDnsFqdn string) (string, error) {
+//HostedServiceClient is used to manage operations on Azure Hosted Services
+type HostedServiceClient struct {
+	client *azure.Client
+}
```

Comment:
What about moving structs to one place?
I agree that we should keep <code>NewClient()</code> here